### PR TITLE
MM-31993: Add flag to disable gossip compression

### PR DESCRIPTION
**Pull request by agnivade**
Full context in the server PR (https://github.com/mattermost/mattermost-server/pull/16701)

https://mattermost.atlassian.net/browse/MM-31993

```release-notes
Add gossip compression flag in the admin console
```

### CODE REVIEW

**Review by nronas — APPROVED** [review]
There are some lint failures but logic wise this LGTM

**Comment by agnivade** [comment]
Gentle ping @calebroseland

File: e2e/cypress/integration/enterprise/system_console/cluster_spec.js [hunk 1]
```diff
@@ -50,4 +50,31 @@ describe('Cluster', () => {
...
+            // * Verify that Gossip Compression is set to true by default
+            cy.get('#EnableGossipCompressiontrue').should('have.attr', 'checked');
+        });
+
+        // # Disable Gossip Compression
+        cy.apiUpdateConfig({ClusterSettings: {
+            Enable: true,
+            UseExperimentalGossip: true,
+            EnableGossipCompression: false,
+        }});
```
Comment (calebroseland) [inline e2e/cypress/integration/enterprise/system_console/cluster_spec.js:72]: Minor/stylistic, but I think it'd be clearer unwrapped 0.5/5
```suggestion
        cy.apiUpdateConfig({
            ClusterSettings: {
                Enable: true,
                UseExperimentalGossip: true,
                EnableGossipCompression: false,
            }
        });
```

**Review by calebroseland — APPROVED** [review]
LGTM, just one suggestion on object style; sorry for the delay!

File: e2e/cypress/integration/enterprise/system_console/cluster_spec.js (hunk 1 above)
Comment (agnivade) [reply e2e/cypress/integration/enterprise/system_console/cluster_spec.js:72]: We should really be either auto-formatting these things, or flagging them via a linter.

**Comment by agnivade** [comment]
Adding @sbishel for admin console changes.

**Review by sbishel — APPROVED** [review]

**Comment by cwarnermm** [comment]
Documentation updated via: https://github.com/mattermost/docs/pull/4368

**Pull request merged**